app/src/main/java/com/example/advanceduiapp/PagerActivity.kt

```diff
@@ -1,12 +1,66 @@
 package com.example.advanceduiapp
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import com.example.advanceduiapp.R
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.isVisible
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentStatePagerAdapter
+import com.example.advanceduiapp.data.PageData
+import com.example.advanceduiapp.databinding.ActivityPagerBinding
+import com.example.advanceduiapp.ui.pager.PagerFragment
+import com.example.advanceduiapp.ui.pager.PagerViewModel
 
 class PagerActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityPagerBinding
+    private val viewModel: PagerViewModel by viewModels()
+    private val adapter = ImagePagerAdapter(supportFragmentManager, listOf())
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_pager)
+        binding = ActivityPagerBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        supportActionBar?.hide()
+
+        viewModel.showMultipleViewRecyclerFragment.observe(this) { showMultipleViewRecycler: Boolean ->
+            if (showMultipleViewRecycler) {
+                swapPagerWithRecycler()
+            }
+        }
+
+        viewModel.imageMap.observe(this){ imageMap: List<PageData> ->
+            updateViewPager(imageMap)
+        }
+    }
+
+    private fun updateViewPager(imageMap: List<PageData>){
+        adapter.update(imageMap)
```

Comment:
this is incorrect. you shouldn't reassign the same adapter each time data updated